doc/source/changes.rst

```diff
@@ -1,6 +1,313 @@
+    - ``instrument_is_enabled``: Returns whether or not an instrument is enabled for the current job.
+ - ``wlauto.core.result``:
+    - Added "classifiers" field to Metric objects. This is a dict mapping classifier names (arbitrary strings) to corresponding values for that specific metrics. This is to allow extensions to add extension-specific annotations to metric that could be handled in a generic way (e.g. by result processors). They can also be set in agendas.
+ - Failed jobs will now be automatically retired.
+ - Implemented dynamic device modules that may be loaded automatically on device initialization if the device supports them.
+ - Added support for YAML configs.
+ - Added ``initialze`` and ``finalize`` methods to workloads.
+ - ``wlauto.core.ExecutionContext``:
+    - Added ``job_status`` property that returns the status of the currently running job.
+
+Fixes/Improvements
+##################
 
+Devices
+~~~~~~~~
+ - ``tc2``: Workaround for buffer overrun when loading large initrd blob.
+ - ``juno``:
+     - UEFI config can now be specified as a parameter.
+     - Adding support for U-Boot booting.
+     - No longer auto-disconnects ADB at the end of a run.
+     - Added ``actually_disconnect`` to restore old disconnect behaviour
+     - Now passes ``video`` command line to Juno kernel to work around a known issue where HDMI loses sync with monitors.
+     - Fixed flashing.
+
+Instruments
+~~~~~~~~~~~
+ - ``fps``: now generates detailed FPS traces as well as report average FPS.
+ - ``trace-cmd``:
+     - Fixed buffer_size_file in trace-cmd for non-Android devices
+     - Reduce starting priority.
+     - Now handles trace headers and thread names with spaces
+ - ``energy_probe``: Added ``device_entry`` parameter.
+ - ``hwmon``:
+     - Sensor discovery is now done only at the start of a run.
+     - Now prints both before/after and mean temperatures.
+ - ``daq``:
+     - Now reports energy
+     - Fixed file descriptor leak
+     - ``daq_power.csv`` now matches the order of labels (if specified).
+     - Added ``gpio_sync``. When enabled, this wil cause the instrument to insert a marker into ftrace, while at the same time setting a GPIO pin high.
+     - Added ``negative_values`` parameter. which can be used to specify how negative values in the samples should be handled.
+     - Added ``merge_channels`` parameter. When set DAQ channel will be summed together.
+     - Workload labels, rather than names, are now used in the "workload" column.
+ - ``cpufreq``:
+     - Fixes missing directories problem.
+     - Refined the availability check not to rely on the top-level cpu/cpufreq directory
+     - Now handles non-integer output in get_available_frequencies.
+ - ``sysfs_extractor``:
+     - Fixed Exception when both device and host paths are empty.
+     - Fixed pulled files verification.
+ - ``perf``:
+     - Updated binaries.
+     - Added option to force install.
+     - ``killall`` is now run as root on Android devices.
```

Comment:
should be "on rooted Android devices" -- as-is gives the impression that will fail on unrooted devices